cluster/azure/util.sh

```diff
@@ -0,0 +1,167 @@
+function azure-ensure-config() {
+    if [[ -z "${AZURE_SUBSCRIPTION_ID:-}" ]]; then
+        echo "AZURE_SUBSCRIPTION_ID must be set"
+        exit 1
+    fi
+    if [[ -z "${AZURE_TENANT_ID:-}" ]]; then
+        echo "AZURE_TENANT_ID must be set"
+        exit 1
+    fi
+    export AZURE_DEPLOY_ID="${AZURE_DEPLOY_ID:-kube-$(date +"%Y%m%d-%H%M%S")}"
+    export AZURE_LOCATION="${AZURE_LOCATION:-westus}"
+    export AZURE_MASTER_SIZE="${AZURE_MASTER_SIZE:-"Standard_A1"}"
+    export AZURE_NODE_SIZE="${AZURE_NODE_SIZE:-"Standard_A1"}"
+    export NODE_COUNT="${NODE_COUNT:-3}"
+    export AZURE_USERNAME="${AZURE_USERNAME:-"kube"}"
+
+    export AZURE_OUTPUT_RELDIR="_deployments/${AZURE_DEPLOY_ID}"
+    export AZURE_OUTPUT_DIR="${DIR}/${AZURE_OUTPUT_RELDIR}"
+    mkdir -p "${AZURE_OUTPUT_DIR}"
+
+    case "${AZURE_AUTH_METHOD:-}" in
+        "client_secret")
+            if [[ -z "${AZURE_CLIENT_ID}" ]]; then
+                echo "AZURE_CLIENT_ID must be set"
+                exit 1
+            fi
+            if [[ -z "${AZURE_CLIENT_SECRET}" ]]; then
+                echo "AZURE_CLIENT_SECRET must be set"
+                exit 1
+            fi
+            ;;
+        "device")
+            ;;
+        "")
+            echo "AZURE_AUTH_METHOD not set, assuming \"device\"."
+            echo " - This will be interactive."
+            echo " - Set AZURE_AUTH_METHOD=client_secret and AZURE_CLIENT_ID/AZURE_CLIENT_SECRET to avoid the prompt"
+            export AZURE_AUTH_METHOD="device"
+            ;;
+        *)
+            echo "AZURE_AUTH_METHOD is an unsupported value: ${AZURE_AUTH_METHOD}"
+            exit 1
+            ;;
+    esac
+}
+
+function azure-deploy(){
+    case "${AZURE_AUTH_METHOD}" in
+        "client_secret")
+            auth_params1="--client-secret=${AZURE_CLIENT_SECRET}"
+            auth_params2="--client-id=${AZURE_CLIENT_ID}"
+            ;;
+        "device")
+            auth_params1=()
+            auth_params2=()
+            ;;
+    esac
+    docker run -it \
+        --user "$(id -u)" \
+        -v "${AZURE_OUTPUT_DIR}:/opt/azkube/${AZURE_OUTPUT_RELDIR}" \
+        colemickens/azkube:v0.0.1 /opt/azkube/azkube deploy \
+            --kubernetes-hyperkube-spec="${AZURE_HYPERKUBE_SPEC}" \
+            --deployment-name="${AZURE_DEPLOY_ID}" \
+            --location="${AZURE_LOCATION}" \
+            --tenant-id="${AZURE_TENANT_ID}" \
+            --subscription-id="${AZURE_SUBSCRIPTION_ID}" \
+            --auth-method="${AZURE_AUTH_METHOD}" "${auth_params1}" "${auth_params2}" \
+            --master-size="${AZURE_MASTER_SIZE}" \
+            --node-size="${AZURE_NODE_SIZE}" \
+            --node-count="${NODE_COUNT}" \
+            --username="${AZURE_USERNAME}" \
+            --output-directory="/opt/azkube/${AZURE_OUTPUT_RELDIR}"
+}
+
+function kube-up {
+    date_start="$(date)"
+    echo "++> AZURE KUBE-UP STARTED: ${date_start}"
+
+    export AZURE_HYPERKUBE_SPEC="gcr.io/google_containers/hyperkube:v1.1.8"
```

Comment:
I'm not 100% sure but I think maybe the right way to do this is
gcr.io/google_containers/hyperkube-amd64:v${K8S_VERSION}

In any event, something parameterized seems better than hard-coded.